src/include/miopen/cat/problem_description.hpp

```diff
@@ -0,0 +1,150 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/problem_description_base.hpp>
+#include <miopen/activ.hpp>
+#include <miopen/tensor.hpp>
+
+#include <cassert>
+#include <string>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace cat {
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(const int32_t xCount_,
+                       const TensorDescriptor* const* xDescs_,
+                       const TensorDescriptor& yDesc_,
+                       int32_t dim_)
+        : xDescs(xDescs_), yDesc(yDesc_), xCount(xCount_), dim(dim_)
+    {
+    }
+
+    const TensorDescriptor& GetXDesc(int i) const
+    {
+        if(i >= xCount)
+        {
+            MIOPEN_THROW(miopenStatusBadParm, "CatForward: Invalid tensor index.");
+        }
+        return *xDescs[i];
+    }
+    const TensorDescriptor& GetYDesc() const { return yDesc; }
+    int32_t GetDim() const { return dim; }
+    int32_t GetXCount() const { return xCount; }
+
+    bool IsSameType() const
```

Comment:
Functions which return bool are intended to return true or false. Its not expected to get an exception in case if false value should be returned. It might be not safe from a caller's perspective.
Also, there are several functions which have same behavior below.